spec/faraday/response/logger_spec.rb

```diff
@@ -38,6 +38,65 @@
     expect(resp.body).to eq('hello')
   end
 
+  context 'without configuration' do
+    let(:conn) do
+      Faraday.new do |b|
+        b.response :logger
+        b.adapter :test do |stubs|
+          stubs.get('/hello') { [200, { 'Content-Type' => 'text/html' }, 'hello'] }
+        end
+      end
+    end
+
+    it 'defaults to stdout' do
+      expect(Logger).to receive(:new).with($stdout).and_return(Logger.new(nil))
+      conn.get('/hello')
+    end
+  end
+
+  context 'with default formatter' do
+    let(:formatter) { instance_double(Faraday::Logging::Formatter, request: true, response: true, filter: []) }
+
+    before { allow(Faraday::Logging::Formatter).to receive(:new).and_return(formatter) }
+
+    it 'delegates logging to the formatter' do
+      expect(formatter).to receive(:request).with(an_instance_of(Faraday::Env))
+      expect(formatter).to receive(:response).with(an_instance_of(Faraday::Env))
+      conn.get '/hello'
+    end
+  end
+
+  context 'with custom formatter' do
+    before do
+      class CustomFormatter < Faraday::Logging::Formatter
+        def initialize(*args)
+          super
+        end
+
+        def request(_env)
+          info 'Custom log formatter request'
+        end
+
+        def response(_env)
+          info 'Custom log formatter response'
+        end
+      end
+    end
+
+    after do
+      Object.send(:remove_const, :CustomFormatter)
+    end
```

Comment:
Setting the class to a local variable should keep it from modifying the global ruby module namespace. What do you think about something like this?

```ruby
  let :formatter_class do
    Class.new(Faraday::Logging::Formatter) do
      # impl
    end
  end
```